coalib/settings/Setting.py

```diff
@@ -133,12 +160,30 @@ def __path__(self, origin=None):
 
         if hasattr(self, "origin") and self.origin != "":
             origin = self.origin
+        if glob_escape_origin:
+            origin = glob_escape(origin)
 
         if origin is None:
             raise ValueError("Cannot determine path without origin.")
 
         return os.path.abspath(os.path.join(os.path.dirname(origin), strrep))
```

Comment:
@sils1297 How about this?

```
        strrep = str(self).strip()
        if os.path.isabs(strrep):
            return strrep

        if hasattr(self, "origin") and self.origin != "":
            origin = self.origin

        if origin is None:
            raise ValueError("Cannot determine path without origin.")

        # We need to get full path before escaping since the full path
        # may introduce unintended glob characters
        origin = os.path.abspath(os.path.dirname(origin))

        if glob_escape_origin:
            origin = glob_escape(origin)

        return os.path.normpath(os.path.join(origin, strrep))
```

We need something at the end to convert `/tmp/./test/*.py` to `/tmp/test/*.py`. And `normpath` does this perfectly.

@AbdealiJK Also I just verified this:
```
73% /tmp ❯ ls -l A
total 0
lrwxrwxrwx. 1 adhityaa adhityaa 8 Mar 19 11:47 foo -> /tmp/X/B
73% /tmp ❯ ls X
B
73% /tmp ❯ ls X/B
73% /tmp ❯ touch /tmp/A/foo/file
72% /tmp ❯ ls X/B
file
72% /tmp ❯ python
Python 2.7.10 (default, Sep  8 2015, 17:20:17) 
[GCC 5.1.1 20150618 (Red Hat 5.1.1-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.path.normpath("/tmp/A/foo/../foo/./file")
'/tmp/A/foo/file'
```
So this basically goes to show that `normpath` does not introduce symlinks (which was the concern you raised yesterday as the symlinked paths might have dangerous characters); it only resolves `.` and `..`

But I agree, `[/]tmp/*.py` will lead to `/[/]tmp/*.py` but that is so weird to use - we could always enforce the condition that absolute paths must start with `/` (but this will rule out stuff like `(/|~/)coala/*.py`; open to any suggestions).